common/v2/features/SendAssets/components/SendAssetsForm.tsx

```diff
@@ -98,18 +104,46 @@ export default function SendAssetsForm({
           const toggleAdvancedOptions = () =>
             setFieldValue('isAdvancedTransaction', !values.isAdvancedTransaction);
 
-          const estimateGasHandler = () => {
-            if (values && values.network) {
-              const provider = new ProviderHandler(values.network);
-              const processedTx = processFormDataToTx(values);
-              if (processedTx) {
-                provider
-                  .estimateGas(processedTx)
-                  .then(data => setFieldValue('gasLimitEstimated', data));
-              }
+          const handleGasEstimate = async () => {
+            if (!values || !values.network) {
+              return;
+            }
+            if (validateTxFields(values)) {
+              return;
+            }
+            const finalTx = processFormDataToTx(values);
+            if (!finalTx) {
+              return;
+            }
+            const gas = await getGasEstimate(values.network, finalTx);
+            setFieldValue('gasLimitEstimated', gas);
+          };
+
+          const handleENSResolve = async (name: string) => {
+            if (!values || !values.network) {
+              return;
+            }
+            setFieldValue('isResolvingNSName', true);
+            const resolvedAddress: string | null = await getResolvedENSAddress(
+              values.network,
+              name
+            );
+            setFieldValue('isResolvingNSName', false);
+            resolvedAddress === null
+              ? setFieldValue('resolvedNSAddress', '0x0')
+              : setFieldValue('resolvedNSAddress', resolvedAddress);
+
+            if (resolvedAddress) {
+              handleGasEstimate();
             }
           };
 
+          const handleFieldReset = () => {
+            setFieldValue('account', undefined);
+            setFieldValue('recipientAddress', '');
+            setFieldValue('amount', '');
+          };
+
           const setAmountFieldToAssetMax = () =>
             // @TODO get asset balance and subtract gas cost
             setFieldValue('amount', '1000');
```

Comment:
These functions could probably be handled by their appropriate custom component. When the component receives a value, onChange or onBlur trigger the function. 

Formik fields can handle their own state, so that we can keep FormData state as minimal as possible, and only use setState/update state when we need to pass up the transaction data. 

https://jaredpalmer.com/formik/docs/api/field#component